compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4647,35 +4645,22 @@ object Types extends TypeUtils {
 
     /** Exists if the tycon is a TypeRef of an alias with an underlying match type.
      *  Anything else should have already been reduced in `appliedTo` by the TypeAssigner.
+     *  May reduce several HKTypeLambda applications before the underlying MatchType is reached.
      */
     override def underlyingMatchType(using Context): Type =
       if ctx.period != validUnderlyingMatch then
         validUnderlyingMatch = if tycon.isProvisional then Nowhere else ctx.period
         cachedUnderlyingMatch = superType.underlyingMatchType
       cachedUnderlyingMatch
 
-    override def tryNormalize(using Context): Type = tycon.stripTypeVar match {
-      case tycon: TypeRef =>
-        def tryMatchAlias = tycon.info match
```

Comment:
The use of a local method is intentional, to keep methods small for JIT.  So you might want to keep that.

Reply:
Does it need be within a `case TypeRef` to be optimal ?

Reply:
No, it just needs to have a small body.  Having it super local just gives you handy references, so you don't have to lift them yourself into method parameters and arguments.